chromadb/test/property/strategies.py

```diff
@@ -39,46 +39,130 @@ def collections(draw) -> Collection:
     """Strategy to generate a set of collections"""
 
     # name = draw(st.from_regex(coll_name_re))
-    name = draw(st.one_of(st.from_regex(_coll_name_re)))
+    name = draw(st.one_of(st.from_regex(_collection_name_re)))
     hypothesis.assume(not _ipv4_address_re.match(name))
     hypothesis.assume(not _two_periods_re.search(name))
 
-    return {"name": name, "metadata": draw(metadata)}
+    return {"name": name, "metadata": draw(collection_metadata)}
+
+
+def one_or_both(strategy_a, strategy_b):
+    return st.one_of(
+        st.tuples(strategy_a, strategy_b),
+        st.tuples(strategy_a, st.none()),
+        st.tuples(st.none(), strategy_b),
+    )
+
+
+@st.composite
+def unique_ids_strategy(draw, count: int):
+
+    ratio = 20
+    strs = count // ratio
+
+    str_results = draw(
+        st.lists(st.text(min_size=1, max_size=64), min_size=strs, max_size=strs, unique=True)
+    )
+
+    # Rotate selections from between the two lists
+    results = []
```

Comment:
For my understanding - whats the motivation for doing a combination of uuids and random ids?

Reply:
UUIDs are fixed length, so this prevents the data generation health check issue with Hypothesis, while still giving good coverage. @levand can say more.

Reply:
Can we just leave a comment explaining this is a workaround.